sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt

```diff
@@ -61,4 +76,77 @@ class SentryNavigationListener @JvmOverloads constructor(
         hint.set(TypeCheckHint.ANDROID_NAV_DESTINATION, destination)
         hub.addBreadcrumb(breadcrumb)
     }
+
+    private fun startTracing(
+        controller: NavController,
+        destination: NavDestination,
+        arguments: Map<String, Any?>
+    ) {
+        if (!isPerformanceEnabled) {
+            return
+        }
+
+        // we can only have one nav transaction at a time
+        if (activeTransaction != null) {
+            stopTracing()
+        }
+
+        if (destination.navigatorName == "activity") {
+            // we do not trace navigation between activities to avoid clashing with activity lifecycle tracing
+            hub.options.logger.log(
+                DEBUG,
+                "Navigating to activity destination, no transaction captured."
+            )
+            return
+        }
+
+        @Suppress("SwallowedException") // we swallow it on purpose
+        var name = destination.route ?: try {
+            controller.context.resources.getResourceEntryName(destination.id)
+        } catch (e: NotFoundException) {
+            hub.options.logger.log(
+                DEBUG,
+                "Destination id cannot be retrieved from Resources, no transaction captured."
+            )
+            return
+        }
+
+        // we add '/' to the name to match dart and web pattern
+        name = "/" + name.substringBefore('/') // strip out arguments from the tx name
+
+        val transaction =
+            hub.startTransaction(name, NAVIGATION_OP, true, hub.options.idleTimeout, true)
```

Comment:
This means we're now creating a new thread on every navigation right? Does that mean we should increase the priority of https://github.com/getsentry/sentry-java/issues/2130 ?

Reply:
yeah we do, but I don't think it's a decider for this issue tbh. The UI transactions is more of a problem than nav events, since you usually navigate less than interacting with screen. I think the usual usage pattern is like going through 5 screens maybe and that's it (open slack, check channel, check threads) that's it kinda :D

Reply:
If you take a mail app for example where you quickly navigate into a mail and back out, there's quite a few transactions.

Reply:
good example, but with your recent fix it would cancel the timer after each navigation event and kill the thread, right? Regardless, I think we should do #2130 rather sooner than later, though I actually don't see a good solution for it right away (which would work both for Android and Java desktop apps)

Reply:
I'll bump prio on it and see if I can get to it sometime.